src/frontend/screens/Library/components/SteamInstall/SteamInstallButton.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useContext, useState } from 'react'
+import { Tooltip, TooltipProps } from '@mantine/core'
+import { faSteam } from '@fortawesome/free-brands-svg-icons'
+import { Alert, Button } from '@hyperplay/ui'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ContextProvider from 'frontend/state/ContextProvider'
+import { useTranslation } from 'react-i18next'
+import { observer } from 'mobx-react-lite'
+import libraryState from 'frontend/state/libraryState'
+import SteamInstallDialog from './SteamInstallDialog'
+import { launch } from 'frontend/helpers'
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { hasStatus } from 'frontend/hooks/hasStatus'
+
+const tooltipProps: Partial<TooltipProps> = {
+  offset: 16,
+  position: 'bottom',
+  withArrow: true,
+  className: 'Tooltip menu',
+  arrowSize: 10,
+  multiline: true,
+  w: 320,
+  transitionProps: { transition: 'fade-down', duration: 150 }
+}
+
+export default observer(function SteamInstallButton() {
+  const { platform, showDialogModal } = useContext(ContextProvider)
+  const [showInstallDialog, setShowInstallDialog] = useState(false)
+  const [showAlert, setShowAlert] = useState<'success' | 'danger' | 'none'>(
+    'none'
+  )
+  const [isInstalling, setIsInstalling] = useState(false)
+  const { status } = hasStatus('steam')
+  const isLaunching = status === 'playing'
+
+  const isMac = platform === 'darwin'
+  const isSteamInstalled =
+    isMac &&
+    libraryState.sideloadedLibrary.find((game) => game.app_name === 'steam')
+
+  const { t } = useTranslation()
+
+  const wineList = useQuery({
+    queryKey: ['alternativeWine'],
+    queryFn: async () => {
+      const response = await window.api.getAlternativeWine()
+      if (!response) return []
+      return response
+    }
+  })
+
+  const isCompatibilityLayerAvailable =
+    wineList.data && wineList.data.length > 0
+
+  const installSteamMutation = useMutation({
+    mutationKey: ['steamInstall'],
+    onSuccess: () => {
+      setShowAlert('success')
+      setTimeout(() => setShowAlert('none'), 5000)
+      setShowInstallDialog(false)
+      setIsInstalling(false)
+    },
+    onError: () => {
+      setShowAlert('danger')
+      setTimeout(() => setShowAlert('none'), 5000)
+      setIsInstalling(false)
+    },
+    onMutate: () => setIsInstalling(true),
+    mutationFn: async () => window.api.installSteamWindows()
+  })
```

Comment:
you don't need isInstalling anymore, you can just do `installSteamMutation.isLoading` it is doing the same.

Reply:
There is not `isLoading` prop though.
It is only status.

Also, if I dont use the isInstalling it doesnt change the state after finishing installing. Same for lunching.